app/models/cms/builtin.rb

```diff
@@ -150,8 +150,8 @@ def to_xml(options = {})
         x.system_name system_name
         x.layout_id layout_id
         unless options[:short]
-          x.draft { |node| node.cdata draft }
-          x.published { |node| node.cdata published }
+          x.draft { |node| draft.present? ? node.cdata(draft) : nil }
```

Comment:
I don't think this works, because `present?` returns false for `""` and `" "`, so the XML response doesn't have a way to distinguish between `""`, `" "` and `nil`. While JSON does.

And now I attach the best cheatsheet ever. You're welcome.
![rails_blank cheatsheet](https://user-images.githubusercontent.com/17052241/223169953-43d1502d-fb16-45ad-8a24-05b9b9f96df0.png)

Reply:
Aaaaand there are no validations for `draft` and `published` that would make `""` and `" "` impossible 😓 
Nil check it is... Updated in 0a2d607.

Reply:
> And now I attach the best cheatsheet ever. You're welcome.

In the battle for the best cheatsheet ever, I'd say this one saved me a couple times in the past:
![Visual_SQL_JOINS_V2](https://user-images.githubusercontent.com/1305223/223177701-561c2003-9efb-4152-ba33-4bff333d38c3.png)

Reply:
That's for weak people, real programmers use big text file instead of a DB engine.